pilot/pkg/model/sidecar.go

```diff
@@ -211,7 +209,7 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *Config, configNamespa
 	// Now collect all the imported services across all egress listeners in
 	// this sidecar crd. This is needed to generate CDS output
 	out.services = make([]*Service, 0)
-	servicesAdded := make(map[string]struct{})
```

Comment:
actually, per some blog, this is more memory efficient than bool :P.

Reply:
How about we make this map of string and service pointer (same pointer in the array?) This way, in the if/else block below, we can cut down the iteration, and simply merge with the existing one in the map, while maintaining the order in the array.

Reply:
> actually, per some blog, this is more memory efficient than bool :P.

Interesting.. Thanks for pointing that. I will change to service pointer